ui/org.eclipse.pde.launching/src/org/eclipse/pde/internal/launching/JUnitLaunchValidationOperation.java

```diff
@@ -0,0 +1,112 @@
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.debug.core.DebugPlugin;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.debug.core.IStatusHandler;
+import org.eclipse.osgi.service.resolver.BundleDescription;
+import org.eclipse.osgi.util.NLS;
+import org.eclipse.pde.core.plugin.IPluginModelBase;
+import org.eclipse.pde.internal.launching.launcher.LaunchValidationOperation;
+
+public class JUnitLaunchValidationOperation extends LaunchValidationOperation {
+
+	private static final Set<String> JUNIT_JUPITER_BUNLDES = new HashSet<>(Arrays.asList(new String[] {"junit-jupiter-api", //$NON-NLS-1$
+			"org.junit.jupiter.api", //$NON-NLS-1$
+			"junit-jupiter-engine", //$NON-NLS-1$
+			"org.junit.jupiter.engine", //$NON-NLS-1$
+	}));
+
+	private static final Object[] EMPTY = new Object[0];
+	private final Map<Object, Object[]> fErrors = new HashMap<>(2);
+
+	public JUnitLaunchValidationOperation(ILaunchConfiguration configuration, Set<IPluginModelBase> models) {
+		super(configuration, models, null);
+	}
+
+	@Override
+	public void run(IProgressMonitor monitor) throws CoreException {
+		super.run(monitor);
+		if (!fErrors.isEmpty()) {
+			fErrors.clear();
+		}
+		validateJunitBundles();
+	}
+
+	private void validateJunitBundles() {
+		try {
+			checkJunitVersion(fLaunchConfiguration, fModels);
+		} catch (CoreException e) {
+			PDELaunchingPlugin.log(e);
+		}
+	}
+
+	@Override
+	public boolean hasErrors() {
+		return super.hasErrors() || fErrors.size() >= 1;
+	}
+
+	@Override
+	public Map<Object, Object[]> getInput() {
+		Map<Object, Object[]> map = super.getInput();
+		map.putAll(fErrors);
+		return map;
+	}
+
+	@SuppressWarnings("restriction")
+	private void checkJunitVersion(ILaunchConfiguration configuration, Set<IPluginModelBase> models) throws CoreException {
+		org.eclipse.jdt.internal.junit.launcher.ITestKind testKind = org.eclipse.jdt.internal.junit.launcher.JUnitLaunchConfigurationConstants.getTestRunnerKind(configuration);
+		if (testKind.isNull()) {
+			return;
+		}
+		switch (testKind.getId()) {
+			case org.eclipse.jdt.internal.junit.launcher.TestKindRegistry.JUNIT5_TEST_KIND_ID -> checkJunit5Version(models);
+		}
+	}
+
+	private void checkJunit5Version(Set<IPluginModelBase> models) throws CoreException {
+		List<String> junit6Bundles = junitBundles(models, 6);
+		if (!junit6Bundles.isEmpty()) {
+			String bundleIds = String.join(System.lineSeparator(), junit6Bundles);
+			String message = NLS.bind(PDEMessages.JUnitLaunchConfiguration_error_JUnit5LaunchJUnit6Runtime, bundleIds);
+			IStatus status = Status.error(message);
+			IStatusHandler statusHandler = DebugPlugin.getDefault().getStatusHandler(status);
```

Comment:
Can you explain to me that the `statusHandler` is doing exactly when present?

Reply:
AFAIK its responsible for showing a dialog on errors. E.g. tests can define their own status handler, that logs instead of showing a dialog (to avoid e.g. tests waiting forever on dialogs, or having to write test code that closes dialogs).

Reply:
When I tried this in a secondoary Eclipse, but it always returned null for me.
Can you make it return something else? If the handler is always null, I suggest to just remove this code-path.